api/proto/teleport/usageevents/v1/usageevents.proto

```diff
@@ -377,6 +377,36 @@ message AssistNewConversationEvent {
   string category = 1;
 }
 
+// AssistAccessRequest is an event that is emitted when a user requests access
+// to a resource via Assist.
+message AssistAccessRequest {
+  // UserName is anonymized user name
+  string user_name = 1;
```

Comment:
There is no need to put the `user_name` field in this message, we should remove it as the information is already carried by the certificate used to connect to the auth.


<details>
  <summary>boring details</summary>
The messages in `api/proto/teleport/usageevents/v1/usageevents.proto` are here for the various clients (tbot, discover, proxy, ...) to report events to the auth. Then, the auth emits batch of events to prehog using messages from `lib/proto/prehog/v1alpha1/teleport.proto`. Those events contain more context, including cluster_name, license_name, and user_name (user_name is event-sepcific while cluster_name and license_name are on the generic event part).

This metadata is filled by the auth itself (user_name gets populated from the user metadata extracted from the certificate used to speak to the auth, this happens [here](https://github.com/gravitational/teleport/pull/30513/files#diff-f9d80bd23db506a0a06486e2812e59ebd25531ee5246d3476278dde911922a87R383-R384)). This avoids data duplication and ensure no mistake can be made when reporting an event (and no client can lie about its metadata).
</details>

Reply:
Removed